extensions/wikia/DesignSystem/scripts/DesignSystemAuthenticationMenu.js

```diff
@@ -0,0 +1,36 @@
+$(function ($) {
+	'use strict';
+
+	var $signInButton = $('#global-navigation-anon-sign-in'),
+		$registerButton = $('#global-navigation-anon-register');
+
+	function onAuthSuccess() {
+		var redirect = this.url.replace(/.*?redirect=([^&]+).*/, '$1');
+		window.location.href = decodeURIComponent(redirect);
+	}
+
+	function openAuthModal(event) {
+		var url = event.currentTarget.href;
+
+		//Prevent opening modal with shift / alt / ctrl / let only left mouse click
+		if (event.which !== 1 || event.shiftKey || event.altKey || event.metaKey || event.ctrlKey) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1067)